darts/utils/data/tabularization.py

```diff
@@ -884,11 +884,25 @@ def _create_lagged_data_by_moving_window(
             # If `start_time` not included in `time_index_i`, can 'manually' calculate
             # what its index *would* be if `time_index_i` were extended to include that time:
             if not is_target_series and (time_index_i[-1] <= start_time):
-                start_time_idx = (
-                    len(time_index_i)
-                    - 1
-                    + (start_time - time_index_i[-1]) // series_i.freq
-                )
+                # Series frequency represents a non-ambiguous timedelta value (not ‘M’, ‘Y’ or ‘y’)
+                if pd.to_timedelta(series_i.freq, errors="coerce") is not pd.NaT:
+                    start_time_idx = (
+                        len(time_index_i)
+                        - 1
+                        + (start_time - time_index_i[-1]) // series_i.freq
+                    )
+                else:
+                    # Create a temporary DatetimeIndex to extract the actual start index.
+                    start_time_idx = (
+                        len(
+                            pd.date_range(
+                                start=time_index_i[0],
```

Comment:
It might be more efficient to generate the index from the end of the series instead of from the beginning and then just add the len(time_index_i) to it?

Reply:
Absolutely, it would make it slightly faster.

The `else` condition will only occur with `DatetimeIndex`, I would rather rely on `pd.date_range` since darts `generate_index` is probably overkill.

Reply:
Yup, I don't know if this is much faster but at least, it looks similar to the other case